src/libtiled/tilelayer.cpp

```diff
@@ -88,20 +148,11 @@ QRegion TileLayer::region(std::function<bool (const Cell &)> condition) const
 {
     QRegion region;
 
-    for (int y = 0; y < mHeight; ++y) {
-        for (int x = 0; x < mWidth; ++x) {
-            if (condition(cellAt(x, y))) {
-                const int rangeStart = x;
-                for (++x; x <= mWidth; ++x) {
-                    if (x == mWidth || !condition(cellAt(x, y))) {
-                        const int rangeEnd = x;
-                        region += QRect(rangeStart + mX, y + mY,
-                                        rangeEnd - rangeStart, 1);
-                        break;
-                    }
-                }
-            }
-        }
+    QMapIterator< QPair<int, int>, Block* > it(mMap);
```

Comment:
That's a good point, however we don't need this container to be sorted and if you use a `QHash` then `QPoint` will work fine.